zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -992,6 +990,7 @@ public void submitRequestNow(Request si) {
             touch(si.cnxn);
             boolean validpacket = Request.isValid(si.type);
             if (validpacket) {
+                setLocalSessionFlag(si);
```

Comment:
I might be missing some info here (Please correct me if I am). So this happening only for requests going through RequestThrottler? And not the requests going through RequestProcessor? (Since this `submitRequestNow` function is being called only in RequestThrottler)
Appreciate if you could explain little bit more. Thanks

Reply:
I'm not quite sure I understand your question here, all the requests will go through RequestThrottler, and then being put into RequestProcessor now.

So no request will be put into RequestProcessor directly, right? Let me know if I didn't get your question.

Reply:
Thanks @lvfangmin . I think I misunderstood. It was good to clarify and I checked little deeper after that. Yes, now I understood all the requests go through `RequestThrottler`. 

And the change looks good to me.  Thanks for fixing it.